pkg/api/v1/types.go

```diff
@@ -855,6 +855,9 @@ type ClusterSpec struct {
 	// EnableOperatingSystemManager enables OSM which in-turn is responsible for creating and managing worker node configuration
 	EnableOperatingSystemManager bool `json:"enableOperatingSystemManager,omitempty"`
 
+	// KubernetesDashboard holds the configuration for kubernetes-dashboard component
+	KubernetesDashboard *kubermaticv1.KubernetesDashboard `json:"kubernetesDashboard,omitempty"`
```

Comment:
I'd use the same naming as the 2 above fields use, `EnableKubernetesDashboard bool`.

Reply:
Initially, I went for that but then restructured it. IMO we should try to be as close as possible to the Cluster type(CR), to be consistent. In general, even `EnableOperatingSystemManager` will be converted into a struct in a similar fashion.

But I don't really have any strong opinions on this. For me, it would just make it more consistent and intuitive.

Reply:
If other fields are planned to be refactored in the same way then I am fine with it.

/lgtm